src/midi/midi_alsa.cpp

```diff
@@ -200,57 +266,61 @@ bool MidiHandler_alsa::Open(const char *conf)
 		return false;
 	}
 
-	my_client = snd_seq_client_id(seq_handle);
-	snd_seq_set_client_name(seq_handle, "DOSBOX");
+	snd_seq_set_client_name(seq_handle, "DOSBox Staging");
 
-	caps = SND_SEQ_PORT_CAP_READ;
-	if (seq_client == SND_SEQ_ADDRESS_SUBSCRIBERS)
+	unsigned int caps = SND_SEQ_PORT_CAP_READ;
+	if (seq.client == SND_SEQ_ADDRESS_SUBSCRIBERS)
 		caps = ~SND_SEQ_PORT_CAP_SUBS_READ;
 
-	my_port = snd_seq_create_simple_port(seq_handle, "DOSBOX", caps,
-	                                     SND_SEQ_PORT_TYPE_MIDI_GENERIC | SND_SEQ_PORT_TYPE_APPLICATION);
-	if (my_port < 0) {
+	output_port = snd_seq_create_simple_port(
+	        seq_handle, "Virtual MPU-401 output", caps,
+	        SND_SEQ_PORT_TYPE_MIDI_GENERIC | SND_SEQ_PORT_TYPE_APPLICATION);
+
+	if (output_port < 0) {
 		snd_seq_close(seq_handle);
 		LOG_MSG("ALSA: Can't create ALSA port");
 		return false;
 	}
 
-	if (seq_client != SND_SEQ_ADDRESS_SUBSCRIBERS) {
-		/* subscribe to MIDI port */
-		if (snd_seq_connect_to(seq_handle, my_port, seq_client, seq_port) < 0) {
-			if (defaultport) {
-				// if port "65:0" (default) try "17:0" as well
-				seq_client = 17;
-				seq_port = 0; // Update reported values
-				if (snd_seq_connect_to(seq_handle, my_port, seq_client, seq_port) < 0) {
-					// Try 128:0 Timidity port as well
-					// seq_client = 128; seq_port = 0; //Update reported values
-					// if(snd_seq_connect_to(seq_handle,my_port,seq_client,seq_port) < 0) {
-					snd_seq_close(seq_handle);
-					LOG_MSG("ALSA: Can't subscribe to MIDI port (65:0) nor (17:0)");
-					return false;
-					// }
-				}
-			} else {
-				snd_seq_close(seq_handle);
-				LOG_MSG("ALSA: Can't subscribe to MIDI port (%d:%d)",
-				        seq_client, seq_port);
-				return false;
-			}
+	if (seq.client != SND_SEQ_ADDRESS_SUBSCRIBERS) {
+		if (snd_seq_connect_to(seq_handle, output_port, seq.client,
+		                       seq.port) == 0) {
+			snd_seq_client_info_t *info = nullptr;
+			snd_seq_client_info_malloc(&info);
+			assert(info);
+			snd_seq_get_any_client_info(seq_handle, seq.client, info);
+			LOG_MSG("ALSA: Connected to MIDI port %d:%d - %s", seq.client,
+			        seq.port, snd_seq_client_info_get_name(info));
+			snd_seq_client_info_free(info);
+			return true;
 		}
 	}
 
-	LOG_MSG("ALSA: Client initialised [%d:%d]", seq_client, seq_port);
-	return true;
+	snd_seq_close(seq_handle);
+	LOG_MSG("ALSA: Can't connect to MIDI port %d:%d", seq.client, seq.port);
+	return false;
```

Comment:
In this message, is it possible to print the port's info string, if we can get it? (kind of like the /ListAll query):

`ALSA: Can't connect to MIDI port 128:0 - Timidity Port 0`

If the port can't even be queried, then that indicates nothing it listening and the user might not understand how the conf option works or needs more help.  In that case (if query-is-null), maybe we could run listAll() for them or do something like:

```
ALSA: Can't connect to MIDI port 128:0.
ALSA: Please run "mixer /listall" to list available MIDI ports.
```

Or we don't want to give them the command, we could suggest:

```
ALSA: Can't connect to MIDI port 128:0.
ALSA: See the MIDI section in README for more help.
```

Or something like that.  MIDI is probably the most complex things users have to deal with, so a bit more help would be nice here.

Reply:
> In this message, is it possible to print the port's info string, if we can get it? (kind of like the /ListAll query):
> 
> `ALSA: Can't connect to MIDI port 128:0 - Timidity Port 0`

In 99% of cases "Can't connect to…" message will be result of using port address that does't exist - usually e.g. Timidity client will change address from 128 to 129 after restart - stuff like that. I can add code to print client name in case user tries to connect to an *output* port of existing synth to cover remaining 1% of cases… but I think it's overcomplicating things a bit.

> If the info query is null, maybe we could run `ListAll()` for them, or simply recommend it:
> 
> ```
> ALSA: Can't connect to MIDI port 128:0.
> ALSA: Please run "mixer /listall" to list available MIDI ports.
> ```

This message will appear only when user selected the port already, so I don't think it's necessary to add more logs. When user leaves the port name empty, we are connecting automatically now :)

I removed the old, broken logic that tried to connect to port 65:0 or 17:0 - we are used to seeing "Can't connect to" message because of that code ;)

> Or something like that. MIDI is probably the most complex things users have to deal with, so a bit more help would be nice here.

We already have info about `mixer /listmidi` in README and in default .conf file. And I intend to add it to `mixer /?`/`mixer --help` in future PR - I think it will be enough :)

Reply:
> We already have info about mixer /listmidi in README and in default .conf file. And I intend to add it to mixer /?/mixer --help in future PR - I think it will be enough :)

I agree @dreamer!